akka-http2-support/src/main/scala/akka/http/impl/engine/http2/ResponseRendering.scala

```diff
@@ -4,12 +4,45 @@
 
 package akka.http.impl.engine.http2
 
+import akka.http.impl.engine.rendering.AbstractHeaderRenderer
+import akka.http.scaladsl.model.headers.Connection
 import akka.http.scaladsl.model.{ ContentType, ContentTypes, HttpHeader, HttpResponse }
 import akka.http.scaladsl.model.http2.Http2StreamIdHeader
 
 import scala.collection.immutable.VectorBuilder
 
 private[http2] object ResponseRendering {
+
+  // best case it can be a stateless single instance like this
+  private val renderer = new AbstractHeaderRenderer[VectorBuilder[(String, String)]] {
+    override protected def render(h: HttpHeader, builder: VectorBuilder[(String, String)]): Unit = {
+      builder += h.lowercaseName → h.value
+    }
+
+    override protected def suppressed(h: HttpHeader, msg: String): Unit = suppressed(h)
+    override protected def suppressed(h: HttpHeader): Unit = println("TODO log suppressed header")
+
+    override protected def renderDefaultServerHeader(builder: VectorBuilder[(String, String)]): Unit = {
+      // TODO render server header efficiently
```

Comment:
these methods will need state and input parameters, so I guess I need to change it into a stream stage or something